integration/test_dataset.py

```diff
@@ -53,8 +56,87 @@ def test_replace_values(self):
             pass
 
         r = ds.follow("table", "limit=10")["data"]
-        assert r[variable.body["id"]] == [1, 2, 3, 4, 5]
-        ds.delete()
+        try:
+            assert r[variable.body["id"]] == [1, 2, 3, 4, 5]
+        finally:
+            ds.delete()
+
+    def test_append_dataset_any_filter(self):
+        _dataset_metadata = {
+            "caseid": {
+                "name": "Case ID",
+                "type": "numeric"
+            },
+            "age": {
+                "name": "Age",
+                "type": 'numeric',
+            },
+            "newssource": {
+                "name": "News source",
+                "type": "multiple_response",
+                "categories": MR_CATS,
+                "subreferences": [{
+                    "name": "Facebook",
+                    "alias": "newssource_1"
+                }, {
+                    "name": "Twitter",
+                    "alias": "newssource_2"
+                }, {
+                    "name": "Google news",
+                    "alias": "newssource_3"
+                }]
+            },
+        }
+        ds = site.datasets.create({
+            'element': 'shoji:entity',
+            'body': {
+                'name': 'test_mr_any',
+                'table': {
+                    'element': 'crunch:table',
+                    'metadata': _dataset_metadata
+                },
+            }
+        }).refresh()
+        ds_to_append = site.datasets.create({
+            'element': 'shoji:entity',
+            'body': {
+                'name': 'test_append_mr_any',
+                'table': {
+                    'element': 'crunch:table',
+                    'metadata': _dataset_metadata
+                },
+            }
+        }).refresh()
+        ds_rows = [
+            ["caseid", "age", "newssource_1", "newssource_2", "newssource_3"],
+            [1, 25, 1, 2, 1],
+            [2, 41, 1, 2, 2],
+            [3, 33, 1, 1, 1]
+        ]
+        ds_to_append_rows = [
+            ["caseid", "age", "newssource_1", "newssource_2", "newssource_3"],
+            [4, 10, 2, 1, 2],
+            [5, 11, 2, 1, 1],
+            [6, 12, 1, 2, 2]
+        ]
+        Importer().append_rows(ds, ds_rows)
+        Importer().append_rows(ds_to_append, ds_to_append_rows)
+        scrunch_dataset = get_mutable_dataset(ds.body.id, site)
+        scrunch_dataset_to_append = get_mutable_dataset(ds_to_append.body.id, site)
+        try:
+            scrunch_dataset.append_dataset(scrunch_dataset_to_append, filter="newssource.any([1])")
```

Comment:
If `nressource` is a MR variable, then the argument inside `any` should be one of its subvariables I'm afraid. This should look like `newsource.any([newssource_1])`